pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -123,7 +114,9 @@ def advance(self, batch, batch_idx):
 
         if self.trainer.lightning_module.automatic_optimization:
             # in automatic optimization, hand over execution to the OptimizerLoop
-            batch_outputs = self.optimizer_loop.run(split_batch, self.get_active_optimizers(batch_idx), batch_idx)
+            # TODO: total_batch_idx here??
```

Comment:
Any thoughts about this TODO? Isn't batch idx safer?

Reply:
forgot to remove this todo. 
yes, need to leave it here as it was like this before. but in the other location where get_active_optimizers is used we are passing in total_batch_idx. that's why I was wondering. 

the choice here depends whether we expect the optimizer frequencies to apply globally across epoch boundaries or be locally defined/rest. 

I tried total batch idx but we don't have any tests that would check this or fail due to a mismatch.